.vuepress/config-path-checker/result-report.js

```diff
@@ -0,0 +1,107 @@
+const chalk = require('chalk')
+const fs = require('fs')
+const configFile = fs.readFileSync('./.vuepress/config.js', 'utf-8').split('\n')
+const LOG_LEVELS = {
+  none: 0,
+  info: 1,
+  warn: 2,
+  error: 3,
+}
+
+function createReportResult({ type, msgFn }) {
+  return {
+    type,
+    list: [],
+    get msg() {
+      return msgFn(this)
+    },
+  }
+}
+
+function convertErrMsg(obj) {
+  return (
+    `\n${
+      obj.type === 'error'
+        ? chalk.red(obj.msg)
+        : obj.type === 'warn'
+        ? chalk.yellow(obj.msg)
+        : obj.msg
+    }\n\n` +
+    obj.list
+      .map(
+        (item) =>
+          `  ${chalk.red(item.errMsg)}: ${item.fullText} ${chalk.gray(
+            `(${item.fileUrl}:${item.line}:${item.column})`
+          )}`
+      )
+      .join('\n') +
+    '\n'
+  )
+}
+
+function displayReport(result, options) {
+  const errorLevels = []
+
+  Object.keys(result).forEach((k) => {
+    /** @type {ReportResult} */
+    const item = result[k]
+    if (!item.list.length) {
+      return
+    }
+
+    const level = LOG_LEVELS[item.type]
+    errorLevels.push(level)
+    if (level > LOG_LEVELS.none) {
+      console[item.type](convertErrMsg(item))
+    }
+  })
+
+  // should not exit if exitLevel is none
+  const exitLevel = LOG_LEVELS[options.exitLevel.toLowerCase()]
+  if (
+    exitLevel !== LOG_LEVELS.none &&
+    errorLevels.find((level) => level >= exitLevel)
+  ) {
+    console.error(chalk.red('Checking failed\n'))
+    process.exit(1)
+  } else {
+    console.info(chalk.green('Checking passed\n'))
+  }
+}
+
+function getLineAndColumn(path) {
+  const found = configFile.reduce((acc, curr, index) => {
+    // regex to match word path and its associated path on the same line
+    const regex = new RegExp('path.*' + path + '[^/].*', 'g')
+    const found = curr.match(regex)
+    if (!(found == null)) {
```

Comment:
Is there another way to write this condition?
Like `if(found)`?

If I understand right, with `!(found == null)` (I tested it on the console):
- if `found` is `null` -> the condition is `false`, because `!true -> false`
- if `found` is not `null` -> the condition is `true`

With my condition:
- if `found` is `null` -> the condition is `false`
- if `found` is not `null` -> the condition is `true`

Reply:
There is in fact other ways and yours is really good. 
The reason I wrote it like this is more for code comprehension than for code compactness.

The `str.match()` function will return a null if a match is not found
```javascript
const paragraph = 'The quick brown fox jumps over the lazy dog. It barked.';
const regex = /notinthestring/g;
const found = paragraph.match(regex);

console.log(found);
```
```bash
> null
```
So I wrote the if in a way that the reader of the code understand that match will return a `null`

Reply:
Ok, if it's for the code comprehension, I took me multiples minutes to understand the condition because of the combination of `null` + `!`.
But if you tell me that is a common way to do it a condition in JS, I'm ok with that and I hope I'll get used to it one day 🤞😂